typha/docker-image/nsswitch.conf

```diff
@@ -1 +1,3 @@
-hosts: files mdns4_minimal [NOTFOUND=return] dns mdns4
+# This is so minimal. We should've built the typha binary with
+# CGO_ENABLED=0 instead. See github.com/projectcalico/calico/issues/6710
```

Comment:
We need to build with CGO enabled since we are relying on the boringssl C library to provide FIPS compatible encryption, which isn't available otherwise. So I think we should remove this comment.

Otherwise the PR looks good to me.

Reply:
Ack. Updated the comment. Updated the commit-msg. Updated the PR-text.